src/k2/tso/service/TSOWorker.cpp

```diff
@@ -51,27 +54,29 @@ seastar::future<> TSOService::TSOWorker::gracefulStop()
     return seastar::make_ready_future<>();
 }
 
-void TSOService::TSOWorker::RegisterGetTSOTimestampBatch()
+seastar::future<std::tuple<Status, dto::GetTimeStampBatchResponse>>
+TSOService::TSOWorker::handleGetTSOTimestampBatch(dto::GetTimeStampBatchRequest&& request)
 {
-    k2::RPC().registerMessageObserver(dto::Verbs::GET_TSO_TIMESTAMP_BATCH, [this](k2::Request&& request) mutable
+    K2LOG_D(log::tsoserver, "request batchsize: {}", request);
+    K2ASSERT(log::tsoserver, request.batchSizeRequested > 0, "request batch size must be greater than 0.");
+
+    try
     {
-        if (request.payload)
-        {
-            uint16_t batchSize;
-            request.payload->read((void*)&batchSize, sizeof(batchSize));
-
-            // TODO: handle exceptions
-            auto response = request.endpoint.newPayload();
-            auto timestampBatch = GetTimestampFromTSO(batchSize);
-            response->write(timestampBatch);
-            return k2::RPC().sendReply(std::move(response), request);
-        }
-        else
-        {
-            K2LOG_E(log::tsoserver, "GetTSOTimestampBatch comes in without request payload.");
-        }
-        return seastar::make_ready_future();
-    });
+        auto timestampBatchGot = GetTimestampFromTSO(request.batchSizeRequested);
+        GetTimeStampBatchResponse response{.timeStampBatch = timestampBatchGot};
+        K2LOG_D(log::tsoserver, "returned timeStampBatch: {}", timestampBatchGot);
+        return RPCResponse(Statuses::S200_OK("OK"), std::move(response));
+    }
+    catch(const TSONotReadyException& e)
+    {
+        K2LOG_E(log::tsoserver, "TSO Not Ready:{}", e.what());

```

Comment:
do we need these errors here?

Reply:
Just want to have a clear distinction on unexpected error and known error case.